napari/layers/shapes/shape_list.py

```diff
@@ -353,7 +382,7 @@ def _update_mesh_vertices(self, index, edge=False, face=False):
     def _update_z_order(self):
         """Updates the z order of the triangles given the z_index list
         """
-        self._z_order = np.argsort(self._z_index)[::-1]
+        self._z_order = np.argsort(self._z_index)
```

Comment:
Currently, all the shapes appear on top of the MIP of the image/volume layer. Is this our fault or a limitation of vispy? I would expect/want the shapes to be in among the binary blobs.

Reply:
that's sort of our fault right now because of layer's concept would be hard to change - and definitely not something for this PR.